server/room-battle.ts

```diff
@@ -892,6 +893,42 @@ export class RoomBattle extends RoomGames.RoomGame {
 		await FS(logpath).mkdirp();
 		await FS(`${logpath}${this.room.getReplayData().id}.log.json`).write(JSON.stringify(logData));
 		// console.log(JSON.stringify(logData));
+		await this.logPostgres(logData);
+	}
+	static logDatabase = new PostgresDatabase();
+	async logPostgres(logData: AnyObject) {
+		if (!Config.usepostgres) return null;
+		const roomid = Number(this.room.roomid.split('-')[2]);
+		const winner = this.endType === 'tie' ? '' : toID(logData.winner);
+		const {p1, p2} = logData;
+		const p1id = toID(p1);
+		const p2id = toID(p2);
+		const loser = winner ? (p1id === winner ? p2id : p1id) : "";
+		const format = Dex.getFormat(this.format).id;
+		const SQL = require('sql-template-strings');
+
+		let rowid;
+		await Rooms.RoomBattle.logDatabase.transaction(async worker => {
```

Comment:
i think it would make more sense for `transaction` to return the result of its callback, and then do `rowid = await Rooms.RoomBattle.logDatabase.transaction(....)`